source/postcard-schema/src/max_size.rs

```diff
@@ -0,0 +1,122 @@
+//! .
+
+use crate::{schema::{DataModelType, DataModelVariant, NamedType}, Schema};
+
+pub const fn max_size<T: Schema>() -> Option<usize> {
+    max_size_nt(T::SCHEMA)
+}
+
+pub const fn max_size_nt(nt: &NamedType) -> Option<usize> {
+    max_size_dmt(nt.ty)
+}
+
+pub const fn max_size_dmt(dmt: &DataModelType) -> Option<usize> {
+    match dmt {
+        DataModelType::Bool => Some(1),
+        DataModelType::I8 => Some(1),
+        DataModelType::U8 => Some(1),
+        DataModelType::I16 => Some(3),
+        DataModelType::I32 => Some(5),
+        DataModelType::I64 => Some(10),
+        DataModelType::I128 => Some(19),
+        DataModelType::U16 => Some(3),
+        DataModelType::U32 => Some(5),
+        DataModelType::U64 => Some(10),
+        DataModelType::U128 => Some(19),
+        DataModelType::Usize => None, // TODO: these don't impl schema and are platform dependent
+        DataModelType::Isize => None, // TODO: these don't impl schema and are platform dependent
+        DataModelType::F32 => Some(4),
+        DataModelType::F64 => Some(8),
+        DataModelType::Char => Some(5), // I think? 1 len + up to 4 bytes
+        DataModelType::String => None,
+        DataModelType::ByteArray => None,
+        DataModelType::Option(nt) => max_size_nt(nt),
+        DataModelType::Unit => Some(0),
+        DataModelType::UnitStruct => Some(0),
+        DataModelType::NewtypeStruct(nt) => max_size_nt(nt),
+        DataModelType::Seq(_) => None,
+        DataModelType::Tuple(nts) | DataModelType::TupleStruct(nts) => {
+            let mut i = 0;
+            let mut ct = 0;
+            while i < nts.len() {
+                let Some(sz) = max_size_nt(nts[i]) else {
+                    return None;
+                };
+                ct += sz;
+                i += 1;
+            }
+            Some(ct)
+        },
+        DataModelType::Map { key, val } => {
```

Comment:
This is wrong, Map is actually basically `Seq((Key, Val))`, this should return None